repoparse.py

```diff
@@ -0,0 +1,294 @@
+from dulwich.repo import Repo
+from os import getcwd, path
+
+from os import makedirs, system
+from shutil import rmtree
+
+from unidiff import PatchSet, UnidiffParseError
+
+from shared import flush_progress_bar, progress, finish_progress, group, s_get_content, s_get_lines, unwrap_bytes_gen_to_str
+
+import multiprocessing
+
+from treesitter_stage import load_treesitter
+from tree_sitter import Language, Parser
+
+@dataclass(init=True)
+class ParsedCommitInfo:
+
+    # Only single parent is allowed
+    parent_sha: str
+    patch_set: Dict[str, str]
+    langs: Dict[str, str]
+    ids: Dict[str, str]
+
+
+@dataclass
+class ParsedCommits:
+
+    commit_data_by_sha: Dict[str, ParsedCommitInfo]
+
+    def __getitem__(self, item: str) -> ParsedCommitInfo:
+        return self.commit_data_by_sha.get(item)
+
+    def __setitem__(self, key: str, value: ParsedCommitInfo):
+        self.commit_data_by_sha[key] = value
+
+
+@dataclass
+class ParsedRepo:
+
+    commits_by_author: Dict[str, ParsedCommits]
+
+    def __getitem__(self, item: str):
+        return self.commits_by_author.get(item)
+
+    def __setitem__(self, key: str, value: ParsedCommits):
+        self.commits_by_author[key] = value
+
+def cleanup_diff(diff):
+    return '\n'.join(map(lambda s: s[1:] if s[0] == '+' else s, diff))
+
+def patch_set_to_json(repopath, patchSet):
+    dict_obj = {}
+    for patched_file in patchSet:
+        changes_list = []
+        dict_obj[path.join(repopath, patched_file.path)] = changes_list
+        for hunk in patched_file:
+            changes_list.extend(map(str, hunk.target_lines()))
+    
+    return dict_obj
+    
+
+def repo_part_to_json(obj):
+    # Python can't pattern match the argument to a method parameter
+    # by type if we need to pass a method into another function (?)
+    if isinstance(obj, ParsedCommitInfo):
+        return {
+            'parent_sha': obj.parent_sha, 
+            'patch_set': obj.patch_set,
+            'langs': obj.langs,
+            'ids': obj.ids
+        }
+    if isinstance(obj, ParsedCommits):
+        return {k if isinstance(k, str) else k.decode(): v for k, v in obj.commit_data_by_sha.items()}
+    if isinstance(obj, ParsedRepo):
+        return {k if isinstance(k, str) else k.decode(): v for k, v in obj.commits_by_author.items()}
+  
+    return obj
+
+def _parse_repo_commits_chunked(repo_path, enry_dict, quiet, chunk_size):
+    print('Starting parallel map...')
+    repo = Repo(repo_path)
+    with multiprocessing.Pool(5) as p:
+        return p.map(_parse_commit_chunk, [(i, repo_path, enry_dict, list(chunk)) for i, chunk in enumerate(group(repo.object_store, chunk_size))])
+
+def _parse_commit_chunk(chunk_tuple): # chunk is an iterable of whatever the repo.object store yields as an iterable
+    i, repo_path, enry_dict, chunk = chunk_tuple
+    print(f'Chunk {i} started...')
+    
+    repo = Repo(repo_path)
+    parsed_repo = ParsedRepo({})
+
+    parser = Parser()
+    LANG_LIST = ['python']
```

Comment:
it's not constant

Reply:
you can initialize tree sitter somewhere else - and create singleton as an option